### PR TITLE
Backport of fixes to 7.6.3

### PR DESCRIPTION
**Pull request by ljacomet**
Fixes
* #25802
* #25781

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/execution/plan/FinalizerGroup.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import com.google.common.collect.SetMultimap;
 
 import javax.annotation.Nullable;
+import java.util.ArrayDeque;
```
Comment (ljacomet) [inline subprojects/core/src/main/java/org/gradle/execution/plan/FinalizerGroup.java:25]: This was using `kotlin.collection.ArrayDeque`
Was it intentional?

That type is not available in 7.x so I replaced it with the `java.util` version here.

File: gradle.properties [hunk 2]
```diff
@@ -26,4 +26,4 @@ systemProp.gradle.internal.testdistribution.queryResponseTimeout=PT20S
 systemProp.org.gradle.kotlin.dsl.precompiled.accessors.strict=true
 
 # Default performance baseline
-defaultPerformanceBaselines=7.6-commit-c6141aed
+defaultPerformanceBaselines=7.6-commit-58171cb5d87
```
Comment (ljacomet) [inline gradle.properties:29]: This is the second change that is not identical to the backported PRs. It uses the commit ID of the commit right before the one making the change, similar to what was done in the original PR.

**Review by ljacomet — COMMENTED** [review]
Hey @ghale,

I highlighted below the two changes I made compared to the original PRs.

**Comment by ljacomet** [comment]
@bot-gradle test RfR

**Comment by ljacomet** [comment]
@bot-gradle test this

**Review by ghale — APPROVED** [review]
LGTM

**Comment by ljacomet** [comment]
@bot-gradle test and merge

**Pull request merged**